templates/http_challenge.j2

```diff
@@ -0,0 +1,15 @@
+server_tokens off;
+
+server {
+    listen 80 default_server;
+    server_name {{ domain_name }};
+
+    location /.well-known/acme-challenge {
+        root /var/www/letsencrypt;
+        try_files $uri $uri/ =404;
+    }
+
+    location / {
+        rewrite ^ https://{{ domain_name }}$request_uri? permanent;
+    }
+}
```

Comment:
I basically like the idea but seems it won't work. Default_server directive will catch only packets which headers won't match existing server directives. Usually you already have these configured, so you'll actually almost never reach this code :/